extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -315,6 +315,12 @@ private PolarisBaseEntity getPolarisBaseEntity(String query) {
         return results.getFirst();
       }
     } catch (SQLException e) {
+      // This look-up is used for checking if the realm is bootstrap or not.
+      // If we have 1 DB per realm it might happen that the realm is not boostrap
+      // at all.
```

Comment:
So why not throw an exception in that case?

Reply:
yes entity null would mean entity not found and this would prompt bootstrap which inturn would run the bootstrap script. Otherwise RTE here is everything broken, i agree this is not an ideal but this is something we have from persistence we expect EntityResult

Reply:
this logic is too obscure IMHO. Would it be preferable to have a dedicated `isRealmBootstrapped(realmId)` method?

Reply:
How about if take this is in my MetaStoreManager refactor ? https://github.com/apache/polaris/issues/1462
I have assigned it to me